quantecon/optimize/scalar_maximization.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+from numba import jit, njit
+
+@njit
+def maximize_scalar(func, a, b, xtol=1e-5, maxiter=500):
```

Comment:
It will be helpful to add `*args` (to pass to `func`).

Reply:
@oyamad thanks I have added this, although unfortunately you need to be quite careful to pass in a tuple and not a scalar, as I can't figure out how to check the type inside a jitted function

Reply:
Sorry I don't see what you mean. Can you elaborate?

And any difference between adding `args=()` and `*args`?

Reply:
In scipy, if something in `args` is passed that is not a tuple (ie. a scalar), the function will convert it to a tuple. I don't seem to be able to get `isinstance` to work inside a jitted function. If you only want to set one fixed argument, you need to pass `args=(y,)` which is somewhat annoying.

I guess we could you `*args` - I was just following scipy's style.

Reply:
I see thanks.

Following exactly scipy's style makes sense, while `*args` looks more Pythonic, allowing passing e.g. `y=5`. I can't say which is better...